sw/airborne/modules/ctrl/ctrl_eff_sched_rot_wing_v3b.c

```diff
@@ -0,0 +1,516 @@
+ * @author D.C. van Wijngaarden <D.C.vanWijngaarden@tudelft.nl>
+ * Crtl effectiveness scheduler for thr rotating wing drone V3
+ */
+
+#include "modules/ctrl/ctrl_eff_sched_rot_wing_v3b.h"
+
+#include "modules/rot_wing_drone/wing_rotation_controller_v3b.h"
+
+#include "modules/actuators/actuators.h"
+
+#include "firmwares/rotorcraft/stabilization/stabilization_indi.h"
+#include "firmwares/rotorcraft/guidance/guidance_indi_hybrid.h"
+
+
+#include "state.h"
+#include "filters/low_pass_filter.h"
+
+// Define initialization values
+float g2_startup[INDI_NUM_ACT] = STABILIZATION_INDI_G2; //scaled by INDI_G_SCALING
+float g1_startup[INDI_OUTPUTS][INDI_NUM_ACT] = {STABILIZATION_INDI_G1_ROLL,
+                                                STABILIZATION_INDI_G1_PITCH, STABILIZATION_INDI_G1_YAW, STABILIZATION_INDI_G1_THRUST
+                                                };
+float rot_wing_side_motors_g1_p_0[2];
+float rot_wing_side_motors_g1_q_90[2] = ROT_WING_SCHED_G1_Q_90; 
+
+// Define polynomial constants for effectiveness values for aerodynamic surfaces
+#ifndef ROT_WING_SCHED_G1_AERO_CONST_P
+#error "ROT_WING_SCHED_AERO_CONST_P should be defined"
+#endif
+
+#ifndef ROT_WING_SCHED_G1_AERO_CONST_Q
+#error "ROT_WING_SCHED_AERO_CONST_Q should be defined"
+#endif
+
+#ifndef ROT_WING_SCHED_G1_AERO_CONST_R
+#error "ROT_WING_SCHED_AERO_CONST_R should be defined"
+#endif
+
+float rot_wing_aerodynamic_eff_const_g1_p[1] = ROT_WING_SCHED_G1_AERO_CONST_P;
+float rot_wing_aerodynamic_eff_const_g1_q[1] = ROT_WING_SCHED_G1_AERO_CONST_Q;
+float rot_wing_aerodynamic_eff_const_g1_r[1] = ROT_WING_SCHED_G1_AERO_CONST_R;
+
+// Define settings to multiply initial control eff scheduling values
+float lift_d_multiplier = 0.7;
+float g1_p_multiplier = 1.;
+float g1_q_multiplier = 1.;
+float g1_r_multiplier = 1.;
+float g1_t_multiplier = 1.;
+
+bool wing_rotation_sched_activated = true;
+bool pusher_sched_activated = true;
+
+float sched_pitch_hover_deg = 0.;
+float sched_pitch_forward_deg = 10.;
+float sched_lower_hover_speed = 7.;
+float sched_upper_hover_speed = 14.;
+
+float pitch_angle_set = 0;
+float pitch_angle_range = 3.;
+
+// Define filters
+#ifndef ROT_WING_SCHED_AIRSPEED_FILTER_CUTOFF
+#define ROT_WING_SCHED_AIRSPEED_FILTER_CUTOFF 0.1
+#endif
+
+Butterworth2LowPass airspeed_lowpass_filter;
+
+// Define scheduling constants
+const float k_elevator[3] = {0.4603,  -4.81466, -28.8464};
+const float k_rudder[3] = {-26.1434, -0.336403, -1.16702 };
+const float k_pusher[2] = {0.007777, -0.67521};
+
+float I_xx = 0.1184;
+float I_yy = 1.1552;
+float I_zz = 1.18;
+const float weight_sched = 6.94;
+
+// Some moved outer loop settings. TODO: make struct
+float pitch_priority_factor = 11.;
+float roll_priority_factor = 10.;
+float thrust_priority_factor = 7.;
+float pusher_priority_factor = 30.;
+
+bool hover_motors_active = true;
+bool bool_disable_hover_motors = false;
+
+float horizontal_accel_weight = 10.;
+float vertical_accel_weight = 10.;
```

Comment:
Wouldn't it be better to make this a little bit more configurable (even if you are probably the only one to ever fly this...) ?

Reply:
rewritten